code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -49,7 +49,6 @@
 	R.speed = 0 // Remove upgrades.
 	R.ionpulse = FALSE
 	R.magpulse = FALSE
-	R.weather_immunities = initial(R.weather_immunities)

```

Comment:
I am unsure why this would not work. Initial is supposed to return the value to its hard-coded preset. Could it be broken when using a list? Anyway, I will adjust this accordingly when I get sprites for https://github.com/tgstation/tgstation/pull/19101. This fix should be fine in the meantime.